app/src/main/java/com/example/dinhh/soundscape/presentation/screens/mixer/MixerFragment.kt

```diff
@@ -0,0 +1,107 @@
+package com.example.dinhh.soundscape.presentation.screens.mixer
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.widget.LinearLayoutManager
+import android.widget.PopupMenu
+import android.support.v7.widget.RecyclerView
+import android.view.*
+import com.example.dinhh.soundscape.R
+import com.example.dinhh.soundscape.common.invisible
+import com.example.dinhh.soundscape.common.visible
+import com.example.dinhh.soundscape.data.Model
+import com.example.dinhh.soundscape.presentation.screens.sounds.SoundAdapter
+import com.example.dinhh.soundscape.presentation.screens.sounds.SoundFragment
+import kotlinx.android.synthetic.main.fragment_mixer.*
+
+class MixerFragment : Fragment(){
+
+    private lateinit var mixerView: RecyclerView
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        // Inflates the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_mixer, container, false)
+        mixerView = view.findViewById(R.id.mixerList) as RecyclerView
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupMixerView()
+
+        //Play created soundscape
+        playAllBtn.setOnClickListener {
+            playAllBtn.invisible()
+            stopAllBtn.visible()
+            for (i in 0 until Model.selectedSounds.size) {
+                Model.selectedSounds[i].sound.start()
+            }
+        }
+
+        //Stop created soundscape
+        stopAllBtn.setOnClickListener {
+            playAllBtn.visible()
+            stopAllBtn.invisible()
+                for (i in 0 until Model.selectedSounds.size) {
+                    Model.selectedSounds[i].sound.pause()
+                    Model.selectedSounds[i].sound.seekTo(0)
+                }
+        }
+
+        //Add sounds
+        addNewMixerItem.setOnClickListener {
+            showPopup()
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        Model.selectedSounds.clear()
+    }
+
+    private fun showPopup() {
+        val popup: PopupMenu?
+        popup = PopupMenu(activity!!.applicationContext, addNewMixerItem, Gravity.END)
+        popup.inflate(R.menu.popup_menu)
+        popup.setOnMenuItemClickListener(PopupMenu.OnMenuItemClickListener { item: MenuItem? ->
+            when (item!!.itemId) {
+                R.id.naturePopupItem -> {
+                    getSoundsFromSelectedCategory(item.title.toString())
```

Comment:
We should put the string "Nature" instead of get it from the item title. It will be a potential issue if you want to change the title or you do the localization and change the language. No worries I will fix it